src/main/java/duke/utility/Storage.java

```diff
@@ -0,0 +1,153 @@
+     * @param filePath The file path to the storage file
+     */
+    public Storage(String filePath) {
+        this.storagePath = Paths.get(filePath);
+    }
+
+    /**
+     * Saves the task to the hard disk.
+     * The task will be parsed to the task's saving format before it is saved to the hard disk.
+     * @param task The task to be saved
+     * @throws DukeException If the FileWriter fails to read the file
+     */
+    public void saveTaskToFile(Task task) throws DukeException {
+        try {
+            FileWriter fw = new FileWriter(storagePath.toString(), true);
+
+            String toBeAppend = Parser.parseForSave(task);
+            fw.write(toBeAppend);
+            fw.close();
+        } catch (IOException e) {
+            throw new FailToReadFileException();
+        }
+    }
+
+    /**
+     * Changes the task's status to done in the hard disk based on the line number specified.
+     * @param line The task line that wanted to be changed
+     * @throws DukeException If the Scanner or FileWriter fails to read the file
+     */
+    public void changeTaskInFile(int line) throws DukeException {
+        try {
+            StringBuilder sb = new StringBuilder();
+            File file = new File(storagePath.toString());
+            Scanner sc = new Scanner(file);
+            int count = 1;
+
+            while (sc.hasNext()) {
+                String taskLine = sc.nextLine() + "\n";
+
+                if (count == line) {
+                    if (taskLine.charAt(4) == '0') {
+                        taskLine = taskLine.replaceFirst("0", "1");
+                    }
+                }
+                sb.append(taskLine);
+                count++;
+            }
+
+            FileWriter fw = new FileWriter(storagePath.toString());
+            fw.write(sb.toString());
+            fw.close();
+        } catch (IOException e) {
+            throw new FailToReadFileException();
+        }
+    }
+
+    /**
+     * Delete a task in the hard disk based on the line number specified.
+     * @param line The task line that wanted to be changed
+     * @throws DukeException If the Scanner or FileWriter fails to read the file
+     */
+    public void deleteTaskInFile(int line) throws DukeException {
+        try {
+            StringBuilder sb = new StringBuilder();
+            File file = new File(storagePath.toString());
+            Scanner sc = new Scanner(file);
+            int count = 1;
+
+            while (sc.hasNext()) {
+                String taskLine = sc.nextLine() + "\n";
+                if (count != line) {
+                    sb.append(taskLine);
+                }
+                count++;
+            }
+
+            FileWriter fw = new FileWriter(storagePath.toString());
+            fw.write(sb.toString());
+            fw.close();
+        } catch (IOException e) {
+            throw new FailToReadFileException();
+        }
+    }
+
+    private void createStorageFile() throws DukeException {
```

Comment:
I noticed many of your methods are private and i like that there is a conscious effort to make them private, that is great coding practice 👍